src/web/wms/src/main/java/org/geoserver/wms/web/data/StyleEditPage.java

```diff
@@ -122,6 +123,10 @@ protected void onStyleFormSubmit() {
             getCatalog().save(style);
             // provide feedback to the user
             styleForm.info("Style saved");
+            // retrieve sld style for non-sld formatted styles on update
+            if ((!SLDHandler.FORMAT.equals(format))) {
+              getCatalog().getResourcePool().getStyle(stylePath);
```

Comment:
This fix seems reasonable, although I am a little surprised it is using getStyle as opposed to some sort of write method - I assume requesting the file in SLD automatically causes it to be rewritten?

Reply:
That is correct.  It won't do the transformation until it attempts to retrieve the style, and writes over the saved sld.